webpack.config.js

```diff
@@ -0,0 +1,66 @@
+const { parse, resolve } = require('path');
+const { sync: glob } = require('fast-glob');
+const WebpackAssetsManifest = require('webpack-assets-manifest');
+const RailsI18nWebpackPlugin = require('@18f/identity-rails-i18n-webpack-plugin');
+
+const mode = process.env.NODE_ENV || 'development';
+const isProduction = mode === 'production';
+const hashSuffix = isProduction ? '-[contenthash:8].digested' : '';
+const isDevServer = !!process.env.WEBPACK_SERVE;
+
+module.exports = /** @type {import('webpack').Configuration} */ ({
+  mode,
+  devtool: 'eval-source-map',
+  devServer: {
+    static: {
+      directory: './public',
+      watch: false,
+    },
+    port: 3035,
+    headers: { 'Access-Control-Allow-Origin': '*' },
+    hot: false,
+  },
+  entry: glob('app/{components,javascript/packs}/*.{js,jsx}').reduce((result, path) => {
+    result[parse(path).name] = resolve(path);
+    return result;
+  }, {}),
+  output: {
+    filename: `[name]${hashSuffix}.js`,
+    chunkFilename: `[name].chunk${hashSuffix}.js`,
+    sourceMapFilename: `[name]${hashSuffix}.js.map`,
+    path: resolve(__dirname, 'public/packs'),
+    publicPath: isDevServer ? 'http://localhost:3035/packs/' : '/packs/',
```

Comment:
What if we updated the Procfile to have like a `${WEBPACK_PORT:3035}` and use `ENV['WEBPACK_PORT']`/ `process.env.WEBPACK_PORT` everywhere else?

Reply:
something like...
```
web: WEBPACK_PORT=${WEBPACK_PORT:-3035} bundle exec rackup config.ru --port ${PORT:-3000} --host ${HOST:-localhost}
worker: bundle exec good_job start
mailcatcher: mailcatcher -f
webpacker: WEBPACK_PORT=${WEBPACK_PORT:-3035} ./bin/webpack-dev-server
```

Reply:
Yeah, that could be a good idea. If we did that, I might want to just remove the `dev` script from `package.json`, to avoid misleading people into thinking it could be run standalone. It's not much of a convenience for Procfile purposes anyways, as the command itself is quite succinct (`yarn webpack serve`).

Reply:
> What if we updated the Procfile to have like a `WEBPACK_PORT=${WEBPACK_PORT:-3035}` and use `ENV['WEBPACK_PORT']`/ `process.env.WEBPACK_PORT` everywhere else?

Updated in 9e5e9991a.